src/pages/docs/concepts-workers.mdx

```diff
@@ -0,0 +1,183 @@
+import { Callout } from "nextra/components"
+import { Card, Cards } from "nextra/components"
+
+# Introduction
+
+Golem is a durable computing platform that makes it simple to build and deploy
+highly reliable distributed systems.
+
+In server-based programming, the fundamental unit of work is the _server_, which
+accepts incoming connections, and handles requests by generating responses.
+
+In Golem, which is a serverless computing platform, the fundamental unit of
+work is the _worker_. A worker is a _running instance_ of a component, with a
+unique identity, which allows addressing specific workers.
+
+Workers are similar to _lambdas_ or _functions_ in other serverless computing
+platforms, but they are far more powerful and expressive.
+
+<Callout type="info">
+  The relationship between a component and a worker is the same as the relationship between an
+  executable and a process: processes are running instances of executables. While the executable
+  contains mostly code, which awaits execution, a process contains both code, as well as dynamic
+  state, which captures work-in-progress.
+</Callout>
+
+# Worker Essentials
+
+The fundamental elements of every worker include:
+
+- **Identity**. Every worker has a unique identity, which allows addressing
+  specific workers.
+- **State**. Every worker has state, including memory, environment variables,
+  and file system.
+- **API**. Every worker has a public API, defined by its component.
+
+These core elements are discussed in the sections that follow.
+
+## Identity
+
+Every worker has a globally unique identity, which is formed from the
+following elements:
+
+- **Component ID**. Every component deployed on Golem has a globally
+  unique identity (UUID), which is assigned by Golem when a new component
+  is uploaded to Golem.
+- **Worker Name**. The worker name can be chosen by you for each worker.
+  If you do not need to address specific workers, you can also use a UUID for
+  the worker name. The only constraint is that worker names must be unique
+  within the scope of their component.
+
+The unique identity of workers allows them to be addressed individually,
+which unlocks many powerful patterns for building distributed systems.
+
+## State
+
+Workers are inherently stateful, in the same way that any running process
+is stateful. Workers have the following stateful elements:
+
+- **Memory**. A worker has in-memory state, such as global variables, stack,
+  and so on, which constantly changes over the life of the worker.
+- **Environment Variables**. A worker has environment variables, which it
```

Comment:
The component itself is not really affecting the environment variables (although the component ID is passed in one). There are three things determining them:
- A set of predefined environment variables (component id, worker name, component version etc)
- User defined environment variables specified **per worker**, when the worker is created (I'm recommending this as a default way of passing configuration to workers, in the language specific guides)
- Inheriting environment variables when implicitly creating new workers via RPC